edb/edgeql/compiler/triggers.py

```diff
@@ -134,10 +146,17 @@ def compile_triggers(
 
         # Process all the types, starting with the base type
         for subtype in sorted(stypes, key=lambda t: t != stype):
+            if defining_trigger_on and subtype.issubclass(
+                    ctx.env.schema, defining_trigger_on):
+                name = str(defining_trigger_on.get_name(ctx.env.schema))
+                raise errors.SchemaDefinitionError(
+                    f"trigger on {name} would need to fire recursively"
```

Comment:
```suggestion
                raise errors.UnsupportedFeatureError(
                    f"trigger {name} is potentially recursive"
```

Reply:
I think "on" is a better preposition for this relationship than "in", but I took the other half of this suggestion.